cmd/deploy/deploy.go

```diff
@@ -618,3 +596,44 @@ func (dc *DeployCommand) cleanUp(ctx context.Context) {
 		oktetoLog.Infof("could not stop local server: %s", err)
 	}
 }
+
+func (dc *DeployCommand) buildImages(ctx context.Context, cfg *corev1.ConfigMap, c kubernetes.Interface, pipelineData *pipeline.CfgData, deployOptions *Options) error {
+	if deployOptions.Build {
+		buildOptions := &types.BuildOptions{
+			EnableStages: true,
+			Manifest:     deployOptions.Manifest,
+			CommandArgs:  deployOptions.servicesToDeploy,
+		}
+		oktetoLog.Debug("force build from manifest definition")
+		if errBuild := dc.Builder.Build(ctx, buildOptions); errBuild != nil {
+			return updateConfigMapStatusError(ctx, cfg, c, pipelineData, errBuild)
+		}
+	} else {
+		if deployOptions.Manifest.Deploy == nil || deployOptions.Manifest.Deploy.ComposeSection == nil || deployOptions.Manifest.Deploy.ComposeSection.Stack == nil {
+			return nil
+		}
+		stackServices := deployOptions.Manifest.Deploy.ComposeSection.Stack.GetServices()
```

Comment:
I think this can't be guaranteed since the services can have a build section or not...